### DISCUSSION TITLE
Post id from url

### DISCUSSION THREAD

**Discussion by MariaJsn** [sbertix/Swiftagram#145] [Implemented ideas] [answered]
Hi Sbertix :)

Is there any way to get post/media id from url?

I want to get the media details/infos from url with Endpoint.Media.summary(for:) endpoint.
Or is there any different endpoint to get media details/infos from url?

Thanks

**Comment by sbertix** [comment, marked as answer]
Hey @MariaJsn, 

Unfortunately there's no way to achieve this in **Swiftagram**. 
I might try and work on this for `5.0.0`, which is still a long way away though.

**Reply by sbertix** [reply]
Quick update: I've added the method in **Swiftagram** last `development` commit. 
As `5.0.0` is **not** even close to being ready for release at this time, though, I suggest adding the code below to your project, if you require the functionality. @MariaJsn 

```swift
func identifier(from url: URL) -> String? {
   // Prepare the `URL`.
   let components = url.pathComponents
   guard let postIndex = components.firstIndex(of: "p"),
      postIndex < components.count-1 else {
         return nil
   }
   let shortcode = components[postIndex+1]
   // Process the shortcode.
   let alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_"
   let set = CharacterSet(charactersIn: alphabet)
   guard shortcode.rangeOfCharacter(from: set.inverted) == nil else {
      return nil
   }
   var identifier: Int64 = 0
   shortcode.forEach { identifier = identifier*64+Int64(alphabet.firstIndex(of: $0)!.utf16Offset(in: alphabet)) }
   return String(identifier)
}
```

**Reply by MariaJsn** [reply]
Thank you very much @sbertix you are the king
